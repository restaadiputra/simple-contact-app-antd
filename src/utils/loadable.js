import React from 'react';
import LoadableComponent from '@loadable/component';
import LoadingSpinner from 'components/loading-spinner';

export default (callback) => {
  return LoadableComponent(callback, {
    ssr: false,
    fallback: <LoadingSpinner />,
  });
};
