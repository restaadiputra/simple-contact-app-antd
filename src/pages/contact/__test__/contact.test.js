import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';

import Contact from '../contact';
import { renderWithStore } from 'test/utils';

test('should render without error', () => {
  // const data = [
  //   {
  //     id: '93ad6070-c92b-11e8-b02f-cbfa15db428b',
  //     firstName: 'Bilbo',
  //     lastName: 'Baggins',
  //     age: 111,
  //     photo: 'testing.jpg',
  //   },
  // ];

  // const { getByText } = renderWithStore(<Contact />);

  // expect(getByText(/contact/i)).toBeInTheDocument();

  // const { container: mobile } = renderWithStore(
  //   <ResponsiveContext.Provider value={{ width: 640 }}>
  //     <Contact data={data} />
  //   </ResponsiveContext.Provider>
  // );
  // expect(mobile).toMatchSnapshot();

  // const { container: desktop } = renderWithStore(
  //   <ResponsiveContext.Provider value={{ width: 1000 }}>
  //     <Contact data={data} />
  //   </ResponsiveContext.Provider>
  // );
  // expect(desktop).toMatchSnapshot();
});
