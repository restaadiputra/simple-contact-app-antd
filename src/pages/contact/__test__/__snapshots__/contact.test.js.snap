// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without error 1`] = `
<div>
  <h2
    class="ant-typography"
  >
    Contact
  </h2>
  <div
    class="ant-list ant-list-split ant-list-loading"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div>
        <div
          class="ant-spin ant-spin-spinning"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
      <div
        class="ant-spin-container ant-spin-blur"
      >
        <div
          style="min-height: 53px;"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-gtssRu gkbNho"
  >
    <button
      class="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg ant-btn-icon-only sc-dlnjPT hrlJGL"
      type="button"
    >
      <span
        aria-label="plus"
        class="anticon anticon-plus"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <defs>
            <style />
          </defs>
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`should render without error 2`] = `
<div>
  <h2
    class="ant-typography"
  >
    Contact
  </h2>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div>
        <div
          class="ant-spin ant-spin-spinning"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
      <div
        class="ant-spin-container ant-spin-blur"
      >
        <div
          class="ant-table ant-table-empty"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
            >
              <table
                style="table-layout: auto;"
              >
                <colgroup />
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell"
                    >
                      Photo
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      First Name
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Last Name
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Age
                    </th>
                    <th
                      class="ant-table-cell"
                      style="text-align: right;"
                    >
                      Action
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-placeholder"
                  >
                    <td
                      class="ant-table-cell"
                      colspan="5"
                    >
                      <div
                        class="ant-empty ant-empty-normal"
                      >
                        <div
                          class="ant-empty-image"
                        >
                          <svg
                            class="ant-empty-img-simple"
                            height="41"
                            viewBox="0 0 64 41"
                            width="64"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="none"
                              fill-rule="evenodd"
                              transform="translate(0 1)"
                            >
                              <ellipse
                                class="ant-empty-img-simple-ellipse"
                                cx="32"
                                cy="33"
                                rx="32"
                                ry="7"
                              />
                              <g
                                class="ant-empty-img-simple-g"
                                fill-rule="nonzero"
                              >
                                <path
                                  d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                />
                                <path
                                  class="ant-empty-img-simple-path"
                                  d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                        <p
                          class="ant-empty-description"
                        >
                          No Data
                        </p>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
